src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -49,7 +49,41 @@ internal Builder()
             }
 
             /// <summary>
-            /// Gets or sets the length of the array.
+            /// Get and sets the length of the internal array.  When set the internal array is 
+            /// is reallocated to the given capacity if it is not already the specified length.
+            /// </summary>
+            public int Capacity
+            {
+                get { return _elements.Length; }
+                set
```

Comment:
The implementation for `List<T>.Capacity` was taken from `ArrayList.Capacity` during the development of .NET 2.0. At the time this property was added, it's unlikely that the `Array.Resize<T>` method existed, hence why it was not used there.

Considering how much simpler my proposed implementation is, I recommend at least giving it full consideration since it would improve the overall size and readability of the code with (from what I've seen so far) no downside.